Render Tile as div when there is no onClick

felixmosh opened the pull request:
This PR renders `div` instead of `button` when `onClick` method is not provided

felixmosh committed:
Render Tile as div when there is no onClick

felixmosh commented:
I will fix tests at the evening...

wojtekmaj commented:
I'm pretty sure Tile has to always be a button, because otherwise you can't select a tile. onClick will always be defined.

felixmosh commented:
Ok, so how can I render a calendar without any selection functionality?

felixmosh commented:
BTW, I'm using only the MonthView in my app, so it can be rendered without `onClick`

wojtekmaj committed:
Update get-user-locale package

wojtekmaj committed:
Update ESLint to 6.x and eslint-config-aribnb to latest

wojtekmaj committed:
v2.19.1

wojtekmaj committed:
Enable & enforce react/sort-comp ESLint rule

wojtekmaj committed:
Remove prefer-destructuring and prefer-spread ESLint rule overwrites

dependabot[bot] committed:
Bump eslint-utils from 1.3.1 to 1.4.2 (#231)

Bumps [eslint-utils](https://github.com/mysticatea/eslint-utils) from 1.3.1 to 1.4.2.
- [Release notes](https://github.com/mysticatea/eslint-utils/releases)
- [Commits](https://github.com/mysticatea/eslint-utils/compare/v1.3.1...v1.4.2)

Signed-off-by: dependabot[bot] <support@github.com>

wojtekmaj committed:
Update Babel to version 7.5.5

Fixes #232

wojtekmaj committed:
v2.19.2

wojtekmaj committed:
Add "Before you continue" section in documentation

wojtekmaj committed:
[breaking] Remove react-lifecycles-compat polyfill (#219)

* [breaking] Remove react-lifecycles-compat polyfill

* [breaking] Bump minimum React version to 16.3

wojtekmaj committed:
[breaking] Implement controlled component support (#217)

* Implement defaultValue, defaultView, defaultActiveStartDate
* Add onViewChange callback
* Rename onActiveDateChange to onActiveStartDateChange
* Prioritize activeStartDate over value when getting default active start date

wojtekmaj committed:
Fix year not being properly localized in th-TH locale, add formatYear custom function (#241)

Fix an issue where Navigation for century view, decade view and year view would not have titles properly localized in th-TH locale
Fix an issue where Decade and Year tiles would not have labels properly localized in th-TH locale
Add support for custom formatYear function
Add new tests for formatting tiles

Fixes #238

wojtekmaj committed:
Update documentation with new props

wojtekmaj committed:
Fix crash when setting an invalid date

wojtekmaj committed:
Add default color for today's tile

Closes #112

wojtekmaj committed:
Implement showDoubleView prop (#248)

* Implement showDoubleView

* Fix tests for Navigation component

* Add unit tests for showDoubleView

wojtekmaj committed:
Upgrade Parcel to ^1.12.4

wojtekmaj committed:
Add .DS_Store to .gitignore

wojtekmaj committed:
Update rimraf

wojtekmaj committed:
Optimize dates utils (#259)

* Add makeGetRange helper

* Add isValidDate helper

* Fixup makeGetRange

* Fix test names

* More compressed formatting of switch

* Add makeGetEdgeOfNeighbor helper

* Add docs

* Add makeGetEdgeOfNeighborMonth helper

* Remove getDifferentMonth function

* Add comments

* Add makeGetEnd helper

* Refactor toYearLabel

wojtekmaj committed:
Add consts, optimize dates utils for size

wojtekmaj committed:
Move tests to *.spec.js (#261)

* Move tests to *.spec.js

* Update .codeclimate.yml

AlbertCarreras committed:
Change defaultProps values to default params on React functional components (#265)

* removed React defaultProps from Navigation functional component

* removed React defaultProps from all functional components

wojtekmaj committed:
Add Alberto Carreras to contributors

wojtekmaj committed:
Fix makeGetEnd noot used in getEndOfDecade

wojtekmaj committed:
Update FUNDING.yml

wojtekmaj committed:
Use @wojtekmaj/date-utils package (#270)

wojtekmaj committed:
Remove TypeScript typings (#272)

* Remove TypeScript typings

Types will be provided by DefinitelyTyped.

* Delete copy-types.js

* Update package.json

wojtekmaj committed:
Create ESM builds (#249)

* Create ESM builds

* Update instructions for nostyle version

wojtekmaj committed:
v3.0.0-beta

wojtekmaj committed:
Add activeStartDate to tileClassName, tileContent

Closes #257

wojtekmaj committed:
Fix unit tests for tileContent

wojtekmaj committed:
[breaking] Add formatMonthYear to Month abbr

wojtekmaj committed:
Add formatLongDate overwrite

wojtekmaj committed:
Add unit tests for other formatters

wojtekmaj committed:
Add views getter to reduce code duplication

wojtekmaj committed:
Simplify extracting props to avoid code duplication

wojtekmaj committed:
Bump minimum React version to 16.3

wojtekmaj commented:
I gave this a thought and this could work, provided that we won't have any onClick passed to tiles, and this at the moment is always happening. Even if drillDown is not available (like on MonthView), there will be onChange passed. That's not good :( We would need to rethink some other areas of this package.

wojtekmaj committed:
Remove code duplication (#277)

wojtekmaj committed:
v3.0.0-beta.2

wojtekmaj committed:
Fix link to 2.x docs

Closes #279

wojtekmaj committed:
Refactor unit tests

wojtekmaj committed:
Add unit tests

wojtekmaj committed:
Fix defaultValue, defaultActiveStartDate and defaultView ignored when calculating initial active start date

wojtekmaj committed:
Add unit tests for derived calendarType value

wojtekmaj committed:
Add cross-env for multi-platform builds

wojtekmaj committed:
v3.0.0-beta.3

wojtekmaj committed:
Simplify logic of drillUp/drillDown (#285)

* Simplify logic of drillUp/drillDown

* Use callback instead of cb variable name

wojtekmaj committed:
Extract variable instead of calling this.valueType twice (#286)

wojtekmaj committed:
Don't repeat the same arguments list twice (#287)

wojtekmaj committed:
Remove mergeFunctions util (#284)

wojtekmaj committed:
[breaking] Remove backwards compatibility with renderChildren (#283)

wojtekmaj committed:
Simplify get value logic (#288)

* Add getValuePiece helper to avoid code repetition

* Add getDetailValue helper to avoid code repetition

* Simplify argument list

* Fix error

* Fix ESLint issues

wojtekmaj committed:
Don't call getLimitedViews when we already have the result (#289)

wojtekmaj committed:
Add getValue helper (#290)

wojtekmaj committed:
v3.0.0-beta.4

wojtekmaj committed:
Expose events in onChange, onClick* functions (#291)

* Expose events in onChange, onClick* functions

Closes #258

* Fix tests

wojtekmaj committed:
Add unit tests

wojtekmaj committed:
Refactor getCalendarTypeFromLocale

wojtekmaj committed:
Extract variable for clarity

wojtekmaj committed:
Fix PropType disallowing strings in WeekNumber

wojtekmaj committed:
[breaking] Don't import styles by default

wojtekmaj committed:
v3.0.0-beta.5

wojtekmaj committed:
Fix falsy prop value overwritten by state value

wojtekmaj committed:
Fix test suite after entry file change

dependabot[bot] committed:
Bump handlebars from 4.1.2 to 4.5.3 (#301)

Bumps [handlebars](https://github.com/wycats/handlebars.js) from 4.1.2 to 4.5.3.
- [Release notes](https://github.com/wycats/handlebars.js/releases)
- [Changelog](https://github.com/wycats/handlebars.js/blob/master/release-notes.md)
- [Commits](https://github.com/wycats/handlebars.js/compare/v4.1.2...v4.5.3)

Signed-off-by: dependabot[bot] <support@github.com>

wojtekmaj committed:
Group tests in describe()

wojtekmaj committed:
Add new tests for activeStartDate change

wojtekmaj committed:
Change mount() to shallow() in tests

wojtekmaj committed:
Fix activeStartDate not being called on value change (#322)

* Fix activeStartDate not being called on value change

Fixes #305

* Fix double setState, improve unit tests

* Refactor, improve unit tests

wojtekmaj committed:
Remove unrelated propTypes from MonthView

wojtekmaj committed:
Fix dateTransform handling dates in year < 100

wojtekmaj committed:
Update @wojtekmaj/date-utils to ^1.0.2

Closes #323

zorfling committed:
Add locale to navigationLabel (#317)

* Add locale to navigationLabel call

* Update navigationLabel docs

wojtekmaj committed:
Reorder arguments

wojtekmaj committed:
v3.0.0-beta.6

wojtekmaj committed:
Specify that an array passed to`value` and `defaultValue` props must contain 2 values

Closes #329

wojtekmaj committed:
v3.0.0

wojtekmaj committed:
Change arrow functions to normal functions

dependabot[bot] committed:
Bump acorn from 5.7.3 to 5.7.4 (#336)

Bumps [acorn](https://github.com/acornjs/acorn) from 5.7.3 to 5.7.4.
- [Release notes](https://github.com/acornjs/acorn/releases)
- [Commits](https://github.com/acornjs/acorn/compare/5.7.3...5.7.4)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

wojtekmaj committed:
Reduce function complexity

wojtekmaj committed:
Update dependencies

wojtekmaj committed:
Update Jest to 25

wojtekmaj committed:
Update cross-env

wojtekmaj committed:
Update ESLint rules

wojtekmaj committed:
Update mkdirp & minimist dependencies

wojtekmaj committed:
Don't call onActiveStartDateChange, onViewChange, onChange if the values didn't change (#344)

* Don't call onActiveStartDateChange, onViewChange, onChange if the values didn't change

* Fix comparing dates

wojtekmaj committed:
v3.0.1

wojtekmaj committed:
Remove duplicate test

wojtekmaj committed:
Refactor processing values (#349)

wojtekmaj committed:
Ensure activeStartDate in calendar views is the beginning of the view

Fixes #345

wojtekmaj committed:
Update Babel to 7.9.0

valtism committed:
Explain hiding icons (#364)

* Explain hiding icons

Explicitly state that setting next and prev attributes to `null` will hide them

* Fix trailing space

Co-authored-by: Wojciech Maj <kontakt@wojtekmaj.pl>

ksoldau committed:
Add tests that ensure the calendar is correct when setting activeStartDate and defaultActiveStartDate (#361)

* Add tests that ensure the dates in a month are lined up with the correct weekdays

* linting

wojtekmaj committed:
Get rid of callIfDefined() in favor of a simple if()

wojtekmaj committed:
Update badges

wojtekmaj committed:
Update eslint-plugin-react-hooks

wojtekmaj committed:
Upgrade Jest to v26

wojtekmaj committed:
Update README.md

wojtekmaj committed:
Add support for allowPartialRange (#351)

* Add support for allowPartialRange

Closes #331

zigotica committed:
Add class names to Navigation label items to allow custom styling (#333)

wojtekmaj committed:
Add Sergi Meseguer to contributors

bywo committed:
Add hoverStart/hoverEnd CSS classes to style hovered date range (#373)

Co-authored-by: Wojciech Maj <kontakt@wojtekmaj.pl>

UtkarshPramodGupta committed:
Fixed re-rendering issue in Calendar component (#300)

* Fixed re-rendering issue in <Calendar/> component

The `onMouseLeave` prop of `<MonthView/>` should be `null` when `selectRange` is not set. It is causing unnecessary re-rendering of the whole `<Calendar/>` component on the hover of `<Weekdays/>` component (child of `<MonthView/>` ) due to fire of the `onMouseOver` event passed in its child `<Flex/>` .

* Remove unnecessary prop

Co-authored-by: Wojciech Maj <kontakt@wojtekmaj.pl>

wojtekmaj committed:
v3.1.0

wojtekmaj committed:
Rewrite basic usage example to hooks

wojtekmaj committed:
Simplify unit test

wojtekmaj committed:
Add unit test for inherited activeStartDate value

wojtekmaj committed:
Fix too complex propTypes for prevLabel, nextLabel

wojtekmaj committed:
Add value to onActiveStartDateChange and onViewChange callbacks arguments (#379)

wojtekmaj committed:
Add event to onChange callback (#388)

wojtekmaj committed:
Update ESLint & eslint-config-airbnb

dependabot[bot] committed:
Bump lodash from 4.17.15 to 4.17.19 (#406)

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.15 to 4.17.19.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.15...4.17.19)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

wojtekmaj committed:
Improve wording

Noor0 committed:
Update `sideEffects` property for webpack 4 (#402)

`Calendar.css` is not bundled in production mode because `sideEffects` is false.
https://stackoverflow.com/questions/49160752/what-does-webpack-4-expect-from-a-package-with-sideeffects-false

wojtekmaj committed:
Add funding to package.json

wojtekmaj committed:
Add link to Travis CI

wojtekmaj committed:
Turn on jest env in test files only

wojtekmaj committed:
Add links to React-Calendar Wiki

dependabot[bot] committed:
Bump yargs-parser from 18.1.1 to 18.1.3 (#427)

Bumps [yargs-parser](https://github.com/yargs/yargs-parser) from 18.1.1 to 18.1.3.
- [Release notes](https://github.com/yargs/yargs-parser/releases)
- [Changelog](https://github.com/yargs/yargs-parser/blob/master/CHANGELOG.md)
- [Commits](https://github.com/yargs/yargs-parser/compare/v18.1.1...v18.1.3)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

wojtekmaj committed:
Allow two-letter locales to be used in test

wojtekmaj committed:
Upgrade Yarn to Yarn Berry (v2) (#430)

* Upgrade Yarn to Yarn Berry (v2)

* Add missing workspace, fix link to parent lib

wojtekmaj committed:
Replace Parcel with Webpack 5 (#432)

wojtekmaj committed:
Fix build command not working on some operating systems

wojtekmaj committed:
Hide previous/next button when prevLabel/nextLabel is set to null (#436)

* Fixed issue where setting prevLabel or nextLabel to null, while making the button invisible, would still render it

Co-authored-by: benomatis <benomatis@gmail.com>

wojtekmaj committed:
Add unit test for wrapper class name

wojtekmaj committed:
Reorder propTypes in alphabetical order

wojtekmaj committed:
Add support for inputRef prop

Closes #384

wojtekmaj committed:
v3.2.0

wojtekmaj committed:
Remove unnecessary Babel override

wojtekmaj committed:
Remove arrow-parens rule override

wojtekmaj committed:
Replace eslint-config-airbnb with eslint-config-wojtekmaj

wojtekmaj committed:
Replace Travis CI with GitHub Actions

wojtekmaj committed:
Remove CodeClimate integration

wojtekmaj committed:
Update eslint-config-wojtekmaj

wojtekmaj committed:
Remove unnecessary cross-env dependency
